### ISSUE TITLE
Uncaught Error: nth not supported on this type function x(a,b,c,d,e){this.kb=a;this.name=b;this.J...

### ISSUE THREAD

**Issue by markaddleman** [mauricioszabo/atom-chlorine#59]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.36.1 x64
**Electron**: 2.0.18
**OS**: Debian GNU/Linux
**Thrown From**: [chlorine](https://github.com/mauricioszabo/atom-chlorine) package 0.1.2

### Stack Trace

Uncaught Error: nth not supported on this type function x(a,b,c,d,e){this.kb=a;this.name=b;this.Jb=c;this.Lc=d;this.eb=e;this.l=2154168321;this.J=4096}

```
At /home/markaddleman/.atom/packages/chlorine/lib/main.js:685

Error: nth not supported on this type function x(a,b,c,d,e){this.kb=a;this.name=b;this.Jb=c;this.Lc=d;this.eb=e;this.l=2154168321;this.J=4096}
    at /packages/chlorine/lib/main.js:108:390)
    at yS (/packages/chlorine/lib/main.js:1427:26)
    at /packages/chlorine/lib/main.js:1432:35
    at Function.b (/packages/chlorine/lib/main.js:1430:205)
    at GE (/packages/chlorine/lib/main.js:685:255)
    at /packages/chlorine/lib/main.js:685:410
    at /packages/chlorine/lib/main.js:670:335
    at /packages/chlorine/lib/main.js:665:298)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
```

### Commands

```
     -0:30.3.0 editor:consolidate-selections (input.hidden-input)
     -0:30.3.0 core:cancel (input.hidden-input)
     -0:29.3.0 grammar-selector:show (input.hidden-input)
  8x -0:27.6.0 core:backspace (input.hidden-input)
     -0:22 editor:consolidate-selections (input.hidden-input)
     -0:22 core:cancel (input.hidden-input)
     -0:21.8.0 inline:clear-current (input.hidden-input)
     -0:21.8.0 editor:consolidate-selections (input.hidden-input)
     -0:21.8.0 core:cancel (input.hidden-input)
     -0:21.3.0 command-palette:toggle (input.hidden-input)
     -0:18.7.0 core:confirm (input.hidden-input)
     -0:18.7.0 chlorine:load-file (input.hidden-input)
     -0:16.3.0 command-palette:toggle (input.hidden-input)
     -0:14.1.0 core:move-down (input.hidden-input)
     -0:13.4.0 core:confirm (input.hidden-input)
     -0:13.3.0 chlorine:go-to-var-definition (input.hidden-input)
```

### Non-Core Packages

```
chlorine 0.1.2 
ink 0.10.5 
parinfer 1.23.0 
teletype 0.13.3 
```

**Comment by mauricioszabo** [comment]
Error in goto var definition, I suppose?

Care to show a code fragment that caused this error? Which var did you try go to definition?

**Comment by markaddleman** [comment]
Hi.  I just started using Chlorine (and Atom) for the first time.  I'm afraid I missed the "steps to reproduce" section int he Github issue.

Yes, this is goto var definition.   I'm working with the project at https://github.com/comnik/clj-3df . I evaluate the file https://github.com/comnik/clj-3df/blob/master/examples/ldbc.clj#L108 and try go to go the schema var.  Chlorine cannot find the definition.  On repeated attempts, I cannot recreate the error above.

**Comment by mauricioszabo** [comment]
Don't worry, just confirming the command I can probably see what's wrong.

There are still lots of parts that are kinda crude in the plug-in, lots of places where when things go wrong it doesn't have any kind of error treatment. This was probably one of these places. I'll keep the issue open until I find out what happened.

**Comment by mauricioszabo** [comment]
Considering you can't recreate the error above, I'm thinking that could be an effect of this: https://github.com/mauricioszabo/repl-tooling/issues/24

To summarize, Clojure's Socket REPL integration currently sends commands separated by newlines. The point is that if you send to Unrepl a command like `(+ 1 2)`, it starts to evaluate _after_ the `)`, so it sends two evaluates: one for the form, and one for a blank line (that'll return `nil`). Chlorine tries to coordinate these evaluations, but sometimes it gets lost.

Maybe it's possible that some "old" evaluate command got mixed in and the coordination didn't work. Will see if I can fix that bug.

**Comment by mauricioszabo** [comment]
I just published version 0.1.5. If this error happens because of an old command waitiing to execute, it'll be fixed.

If you see it happening again, please reopen this issue so we can investigate more? Thanks :)
